### PR TITLE
[DoNOTMerge] Microsoft.Billing - Put for CreateBillingProfile

### PR DESCRIPTION
**Pull request by msft-adrianma**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 1]
```diff
@@ -5708,6 +5772,29 @@
...
+          "type": "string",
+          "format": "date-time"
+        },
+        "endDate": {
+          "description": "End Date.",
+          "type": "string",
+          "format": "date-time"
+        },
+        "additionalDetails": {
+          "description": "This is an free flow string.",
```
Comment (wilcobmsft) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: s/an/a

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 2]
```diff
@@ -5660,6 +5720,10 @@
           "items": {
             "$ref": "#/definitions/InvoiceSection"
           }
+        },
+        "budgetInformation": {
+          "description": "The budget information associated with the billing profile.",
```
Comment (wilcobmsft) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: It's associated with/scoped to the Azure plan, not the billing profile.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json (hunk 1 above)
Comment (wilcobmsft) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: Any reason we don't make this an object, so someone can assign an arbitrary object instead? E.g.:

```
"additionalDetails": {
  "taskOrderId": "123",
  "x": 42
}

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json (hunk 2 above)
Comment (msft-adrianma) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: As per discussion with folks, it was decided to keep this as scoped to BillingProfile and not plan.

**Comment by njuCZ** [comment]
@msft-adrianma May I know the status of this PR? The title says `DoNOTMerge`, and it last several month. Should we review and merge this PR?

**Pull request closed without merging**